### PR TITLE
REF: remove special casing from Index.equals (always dispatch to subclass)

### PR DESCRIPTION
**Pull request by simonjayhawkins**

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -4249,16 +4249,11 @@ def equals(self, other: Any) -> bool:
...
-        if is_object_dtype(self.dtype) and not is_object_dtype(other.dtype):
-            # if other is not object, use other's logic for coercion
-            return other.equals(self)
-
-        if isinstance(other, ABCMultiIndex):
-            # d-level MultiIndex can equal d-tuple Index
-            return other.equals(self)
-
-        if is_extension_array_dtype(other.dtype):
-            # All EA-backed Index subclasses override equals
```
Comment (jbrockmendel) [inline pandas/core/indexes/base.py:4261]: can you add a comment about which cases go through here

**Comment by jbrockmendel** [comment]
small request, otherwise LGTM

**Comment by jreback** [comment]
yep lgtm.

**Comment by simonjayhawkins** [comment]
ok to merge?

**Comment by jreback** [comment]
do we have a test for this?

**Comment by simonjayhawkins** [comment]
we have tests in pandas\tests\indexes\test_base.py and pandas\tests\indexes\multi\test_equivalence.py and

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**